vignettes/nssp_nowcast.Rmd

```diff
@@ -380,95 +376,61 @@ plot_inits
-      from = min(reference_date),
-      to = max(reference_date),
-      by = "day"
-    ),
-    delay = 0:max_delay,
-    fill = list(count = 0)
-  ) |>
-  filter(reference_date + days(delay) <= max(reference_date)) |>
-  select(-report_date) |>
-  pivot_wider(
-    names_from = delay,
-    values_from = count
-  )
-head(rep_tri_df)
+rep_tri <- as_reporting_triangle(training_df,
+  max_delay = max_delay
+)
 ```
 
-We now have a data.frame where each row is a reference date, each column is a delay, and each entry represents the number of cases reported corresponding to each reference date and on each delay.
-The bottom right of the reporting triangle contains NAs to indicate that those observations have yet to be observed.
-As a final pre-processing step, we'll format the reporting triangle as just a matrix, for compatibility with the low-level functions in `baselinenowcast`.
+You can inspect the reporting triangle matrix itself by accessing the $reporting_triangle_matrix element in the object
 ```{r}
-rep_tri <- rep_tri_df |>
-  select(-reference_date) |>
-  as.matrix()
+rep_tri$reporting_triangle_matrix
 ```
 
 ## Specify the `baselinenowcast` model
 
-To specify the number of reference times used for delay and uncertainty estimation, we use an algorithm based on the maximum delay.
-By default, it splits the data evenly (or approximately because of odd numbers of reference times used for training) between data used for delay and uncertainty.
-See the documentation for `?allocate_reference_times` and the "Default Settings" section of the [mathematical model](model_definition.html) vignette for more details.
+Next, we'll specify the number of reference times used for delay and uncertainty estimation as a factor of the maximum delay.
+We'll also specify the proportion of reference times which will be used for delay estimation, with the remaining used for estimating uncertainty.
+We'll set these as the default values, but in a real-world setting we recommend performing an evaluation analysis to identify the optimal amount of training data to be used for each of these tasks.
 ```{r}
-ref_time_allocation <- allocate_reference_times(rep_tri)
-n_history_delay <- ref_time_allocation$n_history_delay
-n_retrospective_nowcasts <- ref_time_allocation$n_retrospective_nowcasts
+ref_time_scale_factor <- 3
+prop_used_for_delay <- 0.5
 ```
 
-The `allocate_reference_times` function returns integer values that indicate the number of reference times to use for delay estimate (`n_history_delay`) and uncertainty estimate (`n_retrospective nowcasts`).
-```{r}
-message(
-  "Number of reference times used for delay estimation: ",
-  n_history_delay
-)
-message(
-  "Number of reference times used for uncertainty estimation: ",
-  n_retrospective_nowcasts
-)
-```
+This means that 25*3, 75 reference times will be used for fitting the model, with about half of them (37) being used for delay estimation and the other half (38) used for uncertainty estimation.
```

Comment:
Still likely want to linkl to the function where this is used i.e internally we cal...